src/library/scala/collection/Iterable.scala

```diff
@@ -481,9 +481,14 @@ trait IterableOps[+A, +CC[_], +C] extends Any with IterableOnce[A] with Iterable
     *  @see [[scala.collection.Iterator]], method `sliding`
     *
     *  @param size the number of elements per group
-    *  @return An iterator producing ${coll}s of size `size`, except the
-    *          last element (which may be the only element) will be truncated
-    *          if there are fewer than `size` elements remaining to be grouped.
+    *  @return An iterator producing ${coll}s of size `size`, except if there
+    *          are between 1 and `size` elements in the original collection,
+    *          in which case the single element is truncated to the original
+    *          collection.
```

Comment:
I don't know if this is clearer. Apparently you mean "the single element (of the resulting iterator)", but it also sounds like it could be "the single element (of the original collection)" which doesn't make sense.

How about just "... in which case, the collection itself is the sole value produced by the iterator."?

Reply:
@hrhino "[...] in the original collection. In that case, an iterator producing the collection itself as its only element is returned."?

Reply:
SGTM.

Reply:
done